File: 47주차/소수찾기.js
function isPrime(num) {
  if (num <= 1) return false;
  for (let i = 2; i < parseInt(num ** 0.5) + 1; i++) {
    if (num % i === 0) return false;
  }
  return true;
}
function solution(numbers) {
  var answer = 0;
  var arr = {};
  let visited = new Array(numbers.length).fill(0);
  function dfs(res) {
    if (res.length > 0) if (isPrime(parseInt(res))) arr[parseInt(res)] = 1;

    for (let i = 0; i < numbers.length; i++) {
      if (!visited[i]) {
        visited[i] = 1;
        dfs(res + numbers[i]);
        visited[i] = 0;
      }
    }
  }

  dfs("");

  arr = Object.keys(arr);
  return arr.length;
}

File: 67주차/소수찾기.py
def isPrime(num):# 소수 판별.
    if num < 2:
        return False
    for i in range(2,int(num**0.5)+1):
        if num % i == 0:
            return False
    return True

def solution(numbers):
    answer = 0
    arr = list(numbers)
    n = len(arr)
    visited = [0]*n
    res = []
    def dfs(tmp):
        
        if 1 <= len(tmp) <= n:# 모든 카드를 사용해서 나올 수 있는 수 구하기.
            res.append(int(tmp))
        
        for i in range(n):
            if not visited[i]:
                visited[i] = 1
                dfs(tmp+arr[i])
                visited[i] = 0
    dfs('')
    res = list(set(res))# 중복 제거.
    
    for v in res:
        if isPrime(v):
            answer += 1
    return answer
